components/properties/src/props.rs

```diff
@@ -105,6 +254,36 @@ impl BidiClass {
     pub const PopDirectionalIsolate: BidiClass = BidiClass(22);
 }
 
+impl_value_getter! {
+    marker: BidiClassNamesV1Marker;
+    impl BidiClass {
+        /// Return a [`PropertyValueLookup`], capable of looking up values
+        /// from strings for the the `Bidi_Class` enumerated property
+        ///
+        /// # Example
+        ///
+        /// ```
+        /// use icu::properties::BidiClass;
+        ///
+        /// let lookup = BidiClass::lookup_values(&icu_testdata::unstable())
+        ///                  .expect("The data should be valid");
+        /// // short name for value
+        /// assert_eq!(lookup.get_strict("AN"), Some(BidiClass::ArabicNumber));
+        /// assert_eq!(lookup.get_strict("NSM"), Some(BidiClass::NonspacingMark));
+        /// // long name for value
+        /// assert_eq!(lookup.get_strict("Arabic_Number"), Some(BidiClass::ArabicNumber));
+        /// assert_eq!(lookup.get_strict("Nonspacing_Mark"), Some(BidiClass::NonspacingMark));
+        /// // name has incorrect casing
+        /// assert_eq!(lookup.get_strict("arabicnumber"), None);
+        /// // loose matching of name
+        /// assert_eq!(lookup.get_loose("arabicnumber"), Some(BidiClass::ArabicNumber));
+        /// // fake property
+        /// assert_eq!(lookup.get_strict("Upside_Down_Vertical_Backwards_Mirrored"), None);
+        /// ```
+        pub fn lookup_values();
```

Comment:
not convinced about this name

Reply:
I think `value_names` to take inspiration from DisplayNames? Except this is the reverse mapping. Maybe `value_name_selector`? Hmm

Reply:
And pair that with `PropertyValueNameSelector<T>`?